Mage/src/main/java/mage/game/events/RemoveCountersEvent.java

```diff
@@ -0,0 +1,33 @@
+package mage.game.events;
+
+import mage.abilities.Ability;
+import mage.cards.Card;
+import mage.players.Player;
+
+public class RemoveCountersEvent extends GameEvent {
+
+    boolean isDamage;
+
+    public RemoveCountersEvent(String name, Card targetCard, Ability source, int amount, boolean isDamage){
+        super(EventType.REMOVE_COUNTERS, targetCard.getId(), source,
+                targetCard.getControllerOrOwnerId(), amount, false);
+
+        if (source != null && source.getControllerId() != null) {
+            setPlayerId(source.getControllerId()); // player who controls the source ability that removed the counters
```

Comment:
Also found weird hack for tokens -- it will be incompatible cause miss controllerId info, must be fixed (add controller info to fake ability):

![shot_240402_122404](https://github.com/magefree/mage/assets/8344157/d8e9ff09-e11d-45a1-837b-e85f6d1a61a5)